snapshot/snapshotfs/upload.go

```diff
@@ -630,7 +641,7 @@ func uploadDirInternal(
 	log.Debugf("finished reading directory %v", dirRelativePath)
 
 	if direrr != nil {
-		return "", fs.DirectorySummary{}, direrr
+		return "", fs.DirectorySummary{}, DirReadError{direrr}
```

Comment:
There was a comment in #35 about special handling for the (top) source directory when it could not be read.
IIUC, this is because `DirReadError` is handled in `processSubdirectories`, which is not in the stack for the top level directory since `Upload()` calls `uploadDir()` => `uploadDirInternal()`.
That is subtle, it may be worth adding a comment in the right place documenting this.

Reply:
Will add some comments. Yes that's right, and by design. My original fix included errors on that root source, but it led to (what I deemed to be) undesirable behavior, i.e. it creates a snapshot anyway, but it's a snapshot of nothing, and it throws errors on restore. Rather than changing the test to expect errors on certain snapshot IDs, I changed the behavior to never ignore errors on the snapshot root.